packages/cli/src/dbt/targets/Databricks/oauth.ts

```diff
@@ -0,0 +1,259 @@
+import { AuthorizationError } from '@lightdash/common';
+import { exec } from 'child_process';
+import * as http from 'http';
+import fetch from 'node-fetch';
+import { generators } from 'openid-client';
+import { URL } from 'url';
+import { promisify } from 'util';
+import GlobalState from '../../../globalState';
+
+const execAsync = promisify(exec);
+
+// Helper function to open browser
+const openBrowser = async (url: string): Promise<void> => {
+    try {
+        const { platform } = process;
+
+        if (platform === 'darwin') {
+            await execAsync(`open "${url}"`);
+        } else if (platform === 'win32') {
+            await execAsync(`start "${url}"`);
+        } else {
+            await execAsync(`xdg-open "${url}"`);
+        }
+    } catch (error) {
+        GlobalState.debug(`> Could not open browser automatically: ${error}`);
+    }
+};
+
+/**
+ * Databricks OAuth tokens result
+ */
+export interface DatabricksOAuthTokens {
+    accessToken: string;
+    refreshToken: string;
+    expiresAt: number; // Unix timestamp in seconds
+}
+
+/**
+ * Perform Databricks U2M OAuth flow
+ * Opens browser for user authentication and exchanges authorization code for tokens
+ * @param host Databricks workspace host
+ * @param clientId OAuth client ID (defaults to 'databricks-cli')
+ */
+export const performDatabricksOAuthFlow = async (
```

Comment:
our U2M code implements full oauth flow, including a redirect server

Databricks offers an automatic authentication method, using their databricks CLI , but when using SDK's authType: 'databricks-oauth', we have no access to the tokens. The SDK handles everything internally, but doesn't persist across CLI process restarts. Each lightdash command =new process = new OAuth flow.

I tested that approach and the UX was worse since we can't have access to the tokens.